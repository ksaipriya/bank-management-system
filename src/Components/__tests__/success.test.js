import React from 'react';
import { render } from '@testing-library/react';
import Success from '../success.js'
import ShallowRenderer from 'react-test-renderer/shallow';
import Adapter from 'enzyme-adapter-react-16';
import * as enzyme from 'enzyme';

enzyme.configure({ adapter: new Adapter() });
describe('Test case for Success Component', () => {
    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<Success/>);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();

    });

    it('check title', () => {
        let wrapper = render(<Success/>)
        const title = wrapper.getByTestId('success')
        expect(title.textContent).toBe('Successfully logged in!')
    })

    it('check apply loan button',()=>{
        const mockCallBack = jest.fn();
        const button = enzyme.shallow(<button onClick={mockCallBack}>Apply Loan</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })

    it('check update details button',()=>{
        const mockCallBack = jest.fn();
        const button = enzyme.shallow(<button onClick={mockCallBack}>Update Details</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    })
})