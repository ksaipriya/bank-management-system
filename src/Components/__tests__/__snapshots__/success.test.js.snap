// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for Success Component matches snapshot 1`] = `
<Container
  fluid={false}
>
  <h4
    data-testid="success"
  >
    Successfully logged in!
  </h4>
  <Form
    inline={false}
  >
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="register"
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="info"
        >
          Apply Loan
        </Button>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="register"
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="info"
        >
          Update Details
        </Button>
      </FormGroup>
    </FormRow>
  </Form>
</Container>
`;
